File: app/src/main/java/com/riders/thelab/ui/recycler/RecyclerClickListener.java
package com.riders.thelab.ui.recycler;

import com.google.android.material.imageview.ShapeableImageView;
import com.riders.thelab.data.remote.dto.artist.Artist;

public interface RecyclerClickListener {
    void onRecyclerClick(Artist artist);

    void onDetailClick(Artist artist, ShapeableImageView sharedImageView, int position);

    void onDeleteClick(Artist artist, int position);
}

File: app/src/main/java/com/riders/thelab/ui/recycler/RecyclerClickListener.kt
package com.riders.thelab.ui.recycler

import com.google.android.material.imageview.ShapeableImageView
import com.riders.thelab.data.remote.dto.artist.Artist

interface RecyclerClickListener {
    fun onRecyclerClick(artist: Artist)

    fun onDetailClick(artist: Artist, sharedImageView: ShapeableImageView, position: Int)

    fun onDeleteClick(artist: Artist, position: Int)
}